### PR TITLE
Support excluding plugin/theme slugs

### PR DESCRIPTION
**Pull request by toolstack**
Allow the user to add a list of plugin/theme slugs to exclude from the data.

### CODE REVIEW

File: admin/class-wp-dev-dashboard-admin.php [hunk 1]
```diff
@@ -891,6 +928,13 @@ public function get_plugins_themes_by_user( $username, $ticket_type = 'plugins'
 			$plugins_themes_by_user = ( 'plugins' == $ticket_type ) ? $data->plugins : $data->themes;
 		}
 
+		// Exclude the slugs the user has told us to.
+		foreach( $plugins_themes_by_user as $key => $value ) {
```
Comment (MickeyKay) [inline admin/class-wp-dev-dashboard-admin.php:932]: Can we switch this over to use a `array_filter()` instead? This is the right use case for that function.

**Review by MickeyKay — CHANGES_REQUESTED** [review]

File: admin/class-wp-dev-dashboard-admin.php (hunk 1 above)
Comment (toolstack) [reply admin/class-wp-dev-dashboard-admin.php:932]: Not easily without either adding the `$exclude_slugs` as a global or a class value as we would not have access to it inside the function used by `array_filter()`.

Comment (MickeyKay) [reply admin/class-wp-dev-dashboard-admin.php:932]: Why, can't we just do something like:

```
$plugins_themes_by_user = array_filter( $plugins_themes_by_user, function( $value ) {
  return in_array( $value->slug, $exclude_slugs ) ? false : true;
});
```

Comment (toolstack) [reply admin/class-wp-dev-dashboard-admin.php:932]: Doesn't work as `$exclude_slugs` isn't in the function scope so it evaluates to null.

**Review by MickeyKay — APPROVED** [review]

**Pull request merged**